In [2]:
import numpy as np
import matplotlib.pyplot as plt
from scipy import integrate
from scipy.special import gamma as gamma 

In [4]:
def tdist(t,nu):
    return  gamma((nu+1)*0.5 )/(gamma(nu*0.5)*np.sqrt(np.pi*nu)) * (1.0 +t**2/nu) **(-(nu+1.0)*0.5)

In [10]:
x_data =[9.8,10.0, 10.4,9.8,10.0,10.2,9.6]
x_mean = 10
n = 7
nu = n -1
alpha = 0.5
alpha2 = 0.025

In [30]:
s2 = np.var(x_data, ddof=1)
np.sqrt(s2)

0.2690370836538196

In [22]:
t_arr = np.linspace(-10,10,1000)
ss = 0
for tval in t_arr:
    prob = integrate.quad(tdist, -np.inf, tval, args=(nu))
    if prob[0] >= 0.975:
        print(t_arr[ss-1])
        break
    ss+=1

2.4324324324324316


Entonces,

$$
t_{(0.025)} = 2.4324
$$

El intervalo de confianza para $\mu$ con $\sigma$-desconocida es:
El intervalo de confianza obtenido es:

$$
IC = [9.75,\ 10.24]
$$